### DISCUSSION TITLE
Coding the "target" attribute on a "link_to"

### DISCUSSION THREAD

**Discussion by belfasthenry** [mojolicious/mojo#1697] [Q&A] [answered]
Within my template I'm coding
```
%= link_to "/view/$view_link" => begin
  %= image $img->{src}, class => 'dirlist', alt => $img->{alt}
% end
```
And it's working fine, generating an anchor tag along these lines:
```
<a href="view/foo/bar/somename.jpg">
  <img alt="DSCI0163.jpg" class="dirlist" src="/foo/thumbnail.jpg">
</a>
```
I want the link to open in a separate tab, which I'd normally do by putting `target="_blank"` in the anchor tag, but I can't find a way of doing it with the `link_to` helper.  Along the lines of the `link_to` documentation in `Mojolicious::Plugin::Taghelpers` I've tried
```
%= link_to "/view/$view_link" => {target=>"_blank"} => begin
  %= image $img->{src}, class => 'dirlist', alt => $img->{alt}
% end
```
And the same with parens round the `target=>` part rather than braces, but both are ignored.  I've looked at the `url_for` sub in `/usr/share/perl5/Mojolicious/Controller.pm` and also in the source of `Mojo::URL` but either it's not there or I can't pick it out of the code.

Can I do this, and if so how?  If not, what?

**Comment by kraih** [comment, marked as answer]
There is no magic, `target` is just another attribute. Just like `class`, `id` or `alt`.

**Reply by belfasthenry** [reply]
Thank you.  I have coded `%= link_to "/view/$view_link" => (target => "_blank") => begin`, which is what I thought I had coded earlier (as described in my initial post) but I must have been mistaken because it now does what I want.

Oddly, "View page source" in Firefox (Ctrl-U) still isn't showing the `target=` attribute on the link; that was throwing me off the scent.  The "Inspect element" tool shows it correctly.  Some cache issue, probably.
